### DISCUSSION TITLE
How to securely setting up a server  as systemd service in ubuntu (or any other linux)

### DISCUSSION THREAD

**Discussion by BeFrankly** [jamulussoftware/jamulus#1154] [Technical Support & Help] [answered]
Hallo out there,
I am going on with testing jamulus on Linux and it work on my linux client. Now I want to set up a server on a Rock Pi hardware that also runs an apache webserver (nextcloud) within a DMZ.
I'm not a newbe in linux and I did get manage to run nextclound an follow most of its security advices. 
But I do not have detail knowledge about setting up a service by hand. Jamulus to my knowledge does not support an installation script for a webservice. I did not find anything in the docs and yes I rtfd (if you would tell me to rtfd).
SO is there a script or a step by step introduction about setting up the server starting with creating a secure, restricted account that runs it?

Thanks for any advice

Frank

**Comment by gilgongo** [comment, marked as answer]
Hi Frank - you can find a description of how to set up a systemd "unit file" here: https://jamulus.io/wiki/Server-Linux ("Create a start script" section). Note that you need to create a non-privileged user with `sudo adduser --system --no-create-home jamulus` first.

In the future, we will have Linux packages that do this automatically, but for now on current versions of Debian and Ubuntu you will need to do it by hand I'm afraid.

**Reply by gilgongo** [reply]
@BeFrankly Out of interest, when you say "I did not find anything in the docs" I'm not surprised, since we have docs all over the place. But can you tell me where you looked first? This may help us improve things :-)

**Reply by BeFrankly** [reply]
Wow, that's astauning. Never had a forum with such fast reply. I will give it a try in a few days. About the documentation: First I want to say that there is good documentation so far but its hard to say, why I sometimes did not find it. Even if I have seen the website of interest some day ago and want to reenter it (not along the browser history, more along the history in my head).
Before I wrote my posting i have been on that site [https://jamulus.io/wiki/Server-Linux](https://jamulus.io/wiki/Server-Linux) but no idea why I did not see this line: **sudo adduser --system --no-create-home jamulus** Maybe I did work to much today or it should have been other color, other size or been bold.

Regards. No further need for help at the moment.
Oh, just tell me how Ritchie Blackmore managed to play that riff in one of the live concerts, you know what I mean :-)

**Comment by BeFrankly** [comment]
Dear @gilgongo , just an advice: I did get manage to run the server now (without the -e switch) but get stuck a while because of problems. I have written the Service Unit file by myself (no copy and paste) and did a mistake by writing the username. Reason for this is, that I created the (lowercase) user "jamulus" this morning and now, in the evening, am writing the Unit file.
The mistake happens because of the mixture of jamulus (lc) and Jamulus (uc). The binary is uppercase as everybody knows. Why do u not let the unitfilename (jamulus.service) and the user to create (jamulus) be uppercase as well?
 
Regards
Frank

**Reply by ann0see** [reply]
The new .deb files for the next release will have a different structure and will be lower case (just as a quick note here)

**Reply by gilgongo** [reply]
> Why do u not let the unitfilename (jamulus.service) and the user to create (jamulus) be uppercase as well?

*nix user names by convention are lower case. I'm afraid I don't know why the compiled version of Jamulus has a capital jay.

**Reply by BeFrankly** [reply]
Thanks. So I'm looking forward to the next release. (Today I have to say, server works on a rock pi as well. Do you have a hardware database or reference list anywhere? So I can give the hardware details to you.

**Reply by ann0see** [reply]
Just keep in mind that the .deb files are Intel/AMD 64 bit only. I couldn't manage ARM builds yet.
